lib/overcommit/hook/pre_push/git_lfs.rb

```diff
@@ -0,0 +1,21 @@
+module Overcommit::Hook::PrePush
+  # Invokes Git LFS command that uploads files tracked by Git LFS to the LFS storage
+  #
+  # @see https://git-lfs.github.com/
+  class GitLfs < Base
+    def run
+      result = execute(['command', '-v', 'git-lfs'])
+      unless result.success?
+        return :warn, 'This repository is configured for Git LFS but \'git-lfs\' ' \
+        'was not found on your path.\nIf you no longer wish to use Git LFS, ' \
+        'disable this hook by removing or setting \'enabled: false\' for GitLFS ' \
+        'hook in your .overcommit.yml file'
+      end
+
+      result = execute(['git', 'lfs', 'pre-push', remote_name, remote_url])
+      return :fail, result.stderr unless result.success?
+
+      :pass
```

Comment:
I don't know of a good way. Would it make sense to instead of actually uploading just check if there is something to upload? The overcommit hook would then output something like 

```
Checking for files that need to be uploaded..........................[GitLFS] FAILED
./some-file.large hasn't been uploaded. Run `git lfs pre-push ... something` to upload. 
```

I'm not a lfs user myself, so this idea might not be possible at all. Or it might just be annoying to have to run the upload manually.